contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -220,33 +236,53 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
     emit FundsWithdrawn(amount, payee);
   }
 
-  /// @notice Sets the top up amount
-  function setTopUpAmount(uint256 topUpAmount) external onlyOwner returns (uint256) {
-    require(topUpAmount > 0, "LinkAvailableBalanceMonitor: invalid linkTokenAddress");
-    return s_topUpAmount = topUpAmount;
+  /// @notice Sets the minimum balance for the given target address
+  function setMinBalance(address target, uint96 minBalance) external onlyOwner {
+    require(target != address(0), "LinkAvailableBalanceMonitor: invalid target address");
+    require(minBalance > 0, "LinkAvailableBalanceMonitor: invalid minBalance");
+    if (!s_targets[target].isActive) {
+      revert InvalidWatchList();
+    }
+    uint256 oldBalance = s_targets[target].minBalance;
+    s_targets[target].minBalance = minBalance;
+    emit BalanceUpdated(target, oldBalance, minBalance);
   }
 
   /// @notice Sets the minimum balance for the given target address
-  function setMinBalance(address target, uint256 minBalance) external onlyOwner returns (uint256) {
-    require(minBalance > 0, "LinkAvailableBalanceMonitor: invalid minBalance");
-    (bool exists, uint256 prevMinBalance) = s_watchList.tryGet(target);
-    if (!exists) {
+  function setTopUpAmount(address target, uint96 topUpAmount) external onlyOwner {
+    require(target != address(0), "LinkAvailableBalanceMonitor: invalid target address");
+    require(topUpAmount > 0, "LinkAvailableBalanceMonitor: invalid topUpAmount");
```

Comment:
usually [custom errors](https://soliditylang.org/blog/2021/04/21/custom-errors/) are preferred whenever possible, because they are substantially smaller and therefore make contract deployments cheaper

Reply:
Moved all requirements to custom errors :)